metricbeat/module/munin/munin.go

```diff
@@ -0,0 +1,109 @@
+package munin
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/joeshaw/multierror"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+const (
+	unknownValue = "U"
+)
+
+// Node connection
+type Node struct {
+	conn net.Conn
+
+	writer io.Writer
+	reader *bufio.Reader
+}
+
+// Connect with a munin node
+func Connect(address string, timeout time.Duration) (*Node, error) {
+	conn, err := net.DialTimeout("tcp", address, timeout)
+	if err != nil {
+		return nil, err
+	}
+	n := &Node{conn: conn,
+		writer: conn,
+		reader: bufio.NewReader(conn),
+	}
+	// Cosume and ignore first line returned by munin, it is a comment
+	// about the node
+	bufio.NewScanner(n.reader).Scan()
+	return n, err
+}
+
+// Close node connection relasing its resources
+func (n *Node) Close() error {
+	return n.conn.Close()
+}
+
+// List of items exposed by the node
+func (n *Node) List() ([]string, error) {
+	_, err := io.WriteString(n.writer, "list\n")
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(n.reader)
+	scanner.Scan()
+	return strings.Fields(scanner.Text()), scanner.Err()
```

Comment:
I see you run here `scanner.Err()` which makes the return value check obsolete before.